.circleci/config.yml

```diff
@@ -2,7 +2,7 @@ version: 2
 jobs:
   build:
     docker:
-      - image: rust:1.54
+      - image: rustlang/rust:nightly
```

Comment:
Do we need nightly for `embedded-hal-async`? Or can we go with a more recent stable version instead?

If we need it, it would probably be good to have a separate CI job which enables the async code, to ensure that the base library always works on stable.

Reply:
Sadly it is still needed for the `type_alias_impl_trait` feature (`generic_assoicated_types` was recently stabilized).
Hopefully we will see `type_alias_impl_trait` stabilized in the new few months :+1: 

I added a second CI job for nightly to maintain both stable & nightly testing.

Reply:
Hmm.  Circle CI is not working the way I thought it would when I add a second job, does something have to be done in the webUI to enable a second job?  I could also port this to github actions.

Reply:
Hmm, it looks correct, not sure why the second job isn't picked up. Feel free to port the config to GitHub actions if you're willing to do that!

Reply:
I updated this PR to use github actions, and opened up #50 to use it on the main branch.  I think github actions has to be enabled for the `master` branch before it will start running on other branches.